filebeat/prospector/log/harvester.go

```diff
@@ -326,7 +344,7 @@ func (h *Harvester) SendStateUpdate() {
 
 	d := util.NewData()
 	d.SetState(h.state)
-	h.forwarder.Outlet.OnEvent(d)
+	h.publishState(d)
```

Comment:
the harvester shouldn't care, but yes. publishState calls `stateOutlet` owned by the prospector. The `stateOutlet` is only closed on `fb.done` or when all harvesters + prospector main loop is done.